ingestion/src/metadata/ingestion/source/database/redshift.py

```diff
@@ -460,6 +468,36 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
             )
         return cls(config, metadata_config)
 
+    def get_table_type(self, schema_name: str, table_name: str) -> TableType:
+        """
+        Fetch the table types and return standard TableType
+        """
+        tabel_fqn = fqn.build(
+            self.metadata,
+            Table,
+            database_name=self.context.database.name.__root__,
+            schema_name=schema_name,
+            table_name=table_name,
+            service_name=self.config.serviceName,
+        )
+        if not self.table_type_map:
+            table_type_query = "SELECT database_name,schema_name, table_name,table_type FROM svv_all_tables"
```

Comment:
reference: https://docs.aws.amazon.com/redshift/latest/dg/r_SVV_ALL_TABLES.html

Reply:
Thanks @ulixius9, clean approach and we handle external tables properly, which we were missing.

Initially I was thinking about preparing this `get_table_type` method for those cases where we have a static answer, such as Athena, where by def tables are external. However, if we do need to run a query to obtain the proper values, that approach does not look valid anymore. Note that with this solution we will be running a query for each table.

Storing the tables and types as we do with `tags` might not be the best approach either, as we might have a bunch of tables. Instead, shall we update:

```
self.inspector.get_table_names(schema_name):
```

to return the table type as well? In those dialects where this does not exist or it is not necessary, we can always update the dialect definition 🤔 

WDYT?